code/src/components/Labyrinth.js

```diff
@@ -0,0 +1,150 @@
+        break;
+      case '1,0':
+        bg =
+          'url(https://images.unsplash.com/photo-1467745870897-1f66443c8c04?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=354&q=80) ';
+        break;
+      case '1,1':
+        bg =
+          'url(https://images.unsplash.com/photo-1428737289360-ca73c271d197?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80)';
+        break;
+      case '0,1':
+        bg =
+          'url(https://images.unsplash.com/photo-1611643892561-9759a32e0c10?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=870&q=80)';
+        break;
+      case '0,2':
+        bg =
+          'url(https://images.unsplash.com/photo-1533693637260-257bd3834be2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=871&q=80)';
+        break;
+      case '0,3':
+        bg =
+          'url(https://images.unsplash.com/photo-1540979730456-da51523895df?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=774&q=80)';
+        break;
+      case '1,3':
+        bg =
+          'url(https://images.unsplash.com/photo-1541429222367-285a893182f2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80)';
+        break;
+      default:
+        bg = 'black';
+    }
+    return bg;
+  };
+  return (
+    loading === false && ( //to make the loading-spinner show in-between fetch
+      <Wrapper
+        style={{
+          background: setBgImage(labyrinth.coordinates),
+          backgroundSize: 'cover',
+        }}
+      >
+        <LabyrinthCard>
+          <Description>{labyrinth.description}</Description>
+          <div>
+            {labyrinth.actions !== undefined && //this is required because the labyrinth is mounted, and username needs to me written and button clicked first, before it should appear
+              labyrinth.actions.map((item) => (
+                <div key={item.description}>
+                  {' '}
+                  {/*key to make .map work*/}
+                  <ItemDescription>{item.description}</ItemDescription>
+                  <Button
+                    onClick={() => {
+                      dispatch(
+                        fetchLabyrinth({
+                          //second fetch
+                          url: 'https://wk16-backend.herokuapp.com/action',
+                          type: 'move',
+                          direction: item.direction,
+                        })
+                      );
+                    }}
+                  >
+                    Go {item.direction}
+                  </Button>
+                </div>
+              ))}
+            {labyrinth.actions.length === 0 && ( //when the labyrinth is ended, the EndPage is mounted
```

Comment:
The fact of rendering the EndPage component based on the length of actions it's super easy to understand!